docs/auth/oidc.md

```diff
@@ -0,0 +1,234 @@
+---
+id: oidc
+title: OIDC provider from scratch
+description: This section shows how to use an OIDC provider from scrath, same steps apply for custom providers.
+---
+
+This section shows how to use an OIDC provider from scratch, same steps apply for custom
+providers. Please note these steps are for using a provider, not how to implement one.
```

Comment:
I think we could mention early on that this generally applies to the OAuth2 provider too, just that the options are a bit different. With that we could change the title of this doc to something like "Generic Auth Providers", keep it at top-level under ´docs/auth´, and call #10303 fixed

Reply:
Have yet to try OAuth2 providers so would like to defer this kind of change until I get experience with it.

Reply:
@Rugvip What makes this OIDC specific is the ID set in the diff in line :88 as well as the config yaml set up at the end of this file, would have to explain that ID more as well as adding multiple yaml config sections, should I do that?

Reply:
@jamieklassen gave me an explanation on the generic OIDC deprecation and why use specific name for the provider, check the update please.